lib/acme/client.rb

```diff
@@ -66,14 +66,19 @@ def connection
     end
   end
 
-  def challenge_from_hash(attributes)
-    case attributes.fetch('type')
+  def challenge_from_hash(arguments)
+    arguments.default_proc = lambda { |_, key| raise "Missing #{key} key" }
```

Comment:
Why not just add a guard check like this :

`%w(type uri token)`.each { |key| raise "Something" unless attributes.key?(key) }